### PR TITLE
Release 2.10: promote experimental features

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

Removes the experimental mark from some features we now consider stable.

#### Which issue(s) this PR fixes or relates to

Ref: [Release 2.10](https://github.com/grafana/mimir/issues/5849)

#### Checklist

- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
 
+
+## 2.10.0-rc.1
+
+### Grafana Mimir
+
+* [FEATURE] The following features are no longer considered experimental. #5872
+  * Ruler storage cache (`-ruler-storage.cache.*`)
+  * Exclude ingesters running in specific zones (`-ingester.ring.excluded-zones`)
+  * Cardinality-based query sharding (`-query-frontend.query-sharding-target-series-per-shard`)
```
Comment (pracucci) [inline CHANGELOG.md:11]: I just noticed that this is not marked as "advanced" but I think it should.

**Review by pracucci — APPROVED** [review]
LGTM. Remember to:

- Remove `experimental` from `RingConfig.Enabled` in `pkg/util/validation/exporter/ring.go`
- Rebuilt auto-generated doc

File: CHANGELOG.md (hunk 1 above)
Comment (colega) [reply CHANGELOG.md:11]: Adjusted in https://github.com/grafana/mimir/pull/5872/commits/0535131e9e598ac50a4d35a6a7b5368e208a0a60

**Review by dimitarvdimitrov — COMMENTED** [review]
Should we also enable some of these features by default now that they're not experimental? I'm thinking about 
* Cardinality-based query sharding
* Cardinality query result caching
* Label names and values query result caching

**Pull request merged**